### ISSUE TITLE
feature suggestion: must use value

### ISSUE THREAD

**Issue by lukeapage** [wix/eslint-plugin-lodash#54]
so we require explicit chains, but I don't like the look of..

```
_.chain(a)
    .map('a')
    .forEach((a) => {})
    .value();
```
I think the code would be clearer and more likely correct if any chains ending in value() are assigned to something, so the above code would have to be
```
var description = _.chain(a)
    .map('a')
    .value();
_.forEach(description, (a) => {})
```
and it would prevent chains not being assigned (e.g. useless code)

wdyt?

btw sorry I know I haven't submitted a pr for my last feature but still slowly switching on your rules on our large codebase...

**Comment by ganimomer** [comment]
Well, since Lodash v4, `forEach` and its kind all end implicit chaining, so you **could** technically write
```js
_(a)
    .map('a')
    .forEach((a) => {})   
```
Which would make this redundant. If you use the `chain-style` with `as-needed`, it would warn against this, and then `no-double-unwrap` would warn against the extra `.value()`.

Also, this would assume that everyone uses each function for its intended use. 
I've seen people use `map`, `filter` and especially `some` for side effects, and in some cases it's too hard for the plugin or Eslint itself to check for that.

**Comment by lukeapage** [comment]
Yes, im aware we could use implicit chaining but i was outvoted, other developers preferred consistency/simplicity over elegance.
A better description might be "avoidCallsOnlyForSideEffects" with an ignore option or somesuch defaulting to forEach.. i guess that doesnt really address my forEach chain issue unless i make the ignores more specific but it does address my side-effects concern.

**Comment by lukeapage** [comment]
And it would be easy to tell if lodash was used only for side effects (basically determine if the parent is a statement), just difficult to tell if it was used for value and side effects (but again possibleb

**Comment by captbaritone** [comment]
Possibly related, I've been thinking that there should be a rule about requiring that the output of `_.each` does not get used, and another that the output of other collection methods DO get used.

This would help prevent people using `_.map` for side effects only.

**Comment by lukeapage** [comment]
Yes, that would be good.

**Comment by ganimomer** [comment]
You make a good point - the whole point of giving options to rules like `chain-style` is not to make assumptions like I just made.
I'd say something like `collection-method-value` would work.

**Comment by lukeapage** [comment]
Good rule, it found 4 cases of confusing code. I am wondering if for my other concern, which is purely stylistic, another rule "disallow-chain-methods"(with option for methods to not allow chaining with) might make sense so I could disallow forEach within a chain ?
